loadtest/control/gencontroller/controller.go

```diff
@@ -96,13 +96,37 @@ func (c *GenController) Run() {
 
 	c.status <- control.UserStatus{ControllerId: c.id, User: c.user, Info: "user started", Code: control.USER_STATUS_STARTED}
 
+	cpaEnabled, resp := control.CustomProfileAttributesEnabled(c.user)
+	if resp.Err != nil {
+		c.sendFailStatus("Failed to retreive CustomProfileAttributesEnabled")
+		return
+	}
+
+	// Create CPA fields
+	if cpaEnabled {
+		actions := map[string]userAction{
+			"createCPAField": {
+				run:        c.createCPAField,
+				frequency:  int(c.config.NumCPAFields),
+				idleTimeMs: 1000,
+			},
+		}
+		c.runActions(actions, func() bool {
+			return st.get(StateTargetCPAFields) >= c.config.NumCPAFields
+		})
+	}
+
 	initActions := []control.UserAction{
 		control.SignUp,
 		c.login,
 		control.GetPreferences,
 		c.createTeam,
 	}
 
+	if cpaEnabled {
+		initActions = append(initActions, c.createCPAValues)
```

Comment:
As this need to happen after `createCPAFields` and `createCPAFields` should happen after the `initActions`, I'm afraid we'll have to run two independent blocks. `createCPAValues` can be called once, though, no need to wrap it in `c.runActions`, I think.